### ISSUE TITLE
Covered/bundled cables?

### ISSUE THREAD

**Issue by LemmaEOF** [elytra/InfraRedstone-Fabric#6]
I'm thinking about how I want to expand InRed, and of course one of the most obvious possibilities is to add cable covering/bundling, so people can put two InRed cables next to each other and have them not connect, and so you can carry a lot of different colors in the same blockspace. Since wires are dumb, covers and bundles won't need to carry much information about themselves except for what color they are, but it means our search logic will have to become a *lot* more complex to adapt. A few things to think about:
- How do we specify what color a search is currently focusing on? How do we go from non-covered to covered wire?
- How do we deal with a situation of multiple colors connecting to the same un-covered cable? Should we add a specific, separate search method that focuses on colored wire?
- Does it feel like this is ultimately a worthwhile addition to InRed? Is there some way we could do a similar thing without the standard RedPower-style bundling system, and would that be better?

**Comment by kitlith** [comment]
about alternatives: you could do stuff with what project red calls 'null cells', where you can cross wires over each other using a single tile without any delay. Something that might be nice if you go this route is the same but for turning corners.

**Comment by LemmaEOF** [comment]
I dunno if a null cell would really fix the primary issue a cover system tries to solve, where you have two devices next to each other and want them to receive different signals. You can use diodes for that right now like you'd use vanilla repeaters, but that still adds delay into the circuit that's not always welcome.

**Comment by kitlith** [comment]
yes, I'd still recommend having at least some sort of covers *or other way to seperate* -- for instance, with project red you can also use microblocks to block off connections in a certain direction. I used this when colored cable wasn't available (framed bundled cable doesn't have any colors to prevent signal crossing, so I used covers to keep it compact)

**Comment by LemmaEOF** [comment]
@falkreon, I'd really love to hear your input on this sometime. You originally called the move from a 0-63 value to a 6-bit value in the presentation, so I wanna know if you think it's a good idea to add bundling on top of that.

**Comment by kitlith** [comment]
it may also be worth considering that if you do bundling, that maybe you don't need the standard 16 lines. e.g. 4-8 lines (or 6 if you want to continue with patterns of 6) still gives you more individual signals than 16 1 bit lines.

at the same time, you get more signals the more bundling you allow. so... yay tradeoffs.
